crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -506,6 +506,12 @@ pub fn handle_rename(world: WorldSnapshot, params: RenameParams) -> Result<Optio
         .into());
     }
 
+    // Only rename to valid identifiers
+    let tokens = tokenize(&params.new_name);
+    if tokens.len() != 1 || tokens[0].kind != SyntaxKind::IDENT {
```

Comment:
Really nice solution, I :heart: this, thanks! But let's move this to `Analysis`, so that we can test it. Probably just

```rust
impl Analysis {
    fn is_valid_name(&self, &str) -> bool
}
```

to avoid thinking to much about proper design